driver-core/src/test/functional/com/mongodb/internal/async/client/ChangeStreamProseTest.java

```diff
@@ -178,11 +178,15 @@ private void insertOneDocument() {
         return futureResult(callback);
     }
 
-    private void setFailPoint(final String command, final int errCode) {
+    private void setFailPoint(final String command, final int errCode, final boolean addResumableChangeStreamError) {
```

Comment:
This seems hacky and hard to understand when read from the call site because I see `setFailPoint("getMore", 10107, true)` and I don't know what the `true` means. I can think of two options:

1. Add a new method like `setResumableChangeStreamErrorFailPoint`. This is a little better in terms of readability, but could get unwieldy quickly if we add new labels in the future, which seems to be the general direction drivers are going in.

2. For Go, I added a new type that represented a fail point and could be serialized into BSON. We now use that type to de-serialize fail point definitions in spec test files and added a function like `setFailPoint(FailPoint fp)` so the whole process requires no manual logic. It's definitely out of scope here, but maybe a JAVA ticket to do this in the future is a good idea?

Reply:
I have added a `FailPoint` class to store the fail point information.